conf/tls/tls_config.go

```diff
@@ -0,0 +1,125 @@
+	Key        NamedFileContent `help:"Private PEM-encoded key to use for ${tls_target}. Only use if unsetting --key-pair." placeholder:"FILE"`
+	Cert       NamedFileContent `help:"Public PEM-encoded cert to use for ${tls_target}. Only use if unsetting --key-pair." placeholder:"FILE"`
+	ClientAuth string           `help:"Client certificate authentication mode, maps to the tls ClientAuthType"`
+}
+
+var clientAuthTypeMap = map[string]tls.ClientAuthType{
+	"NoClientCert":               tls.NoClientCert,
+	"RequestClientCert":          tls.RequestClientCert,
+	"RequireAnyClientCert":       tls.RequireAnyClientCert,
+	"VerifyClientCertIfGiven":    tls.VerifyClientCertIfGiven,
+	"RequireAndVerifyClientCert": tls.RequireAndVerifyClientCert,
+	"":                           tls.RequireAndVerifyClientCert, // The default value if nothing is provided by the configuration
+}
+
+// NamedFileContent is a custom type that is similar to kong.NamedFileContentFlag but includes an error object.
+type NamedFileContent struct {
+	Filename string
+	Contents []byte
+	Error    error // The error returned from ioutil.ReadFile when decoding the corresponding flag.
+}
+
+// Decode is the custom kong decoder that is similar to kong.NamedFileContentFlag's decode implementation,
+// but stores the error from ioutil.ReadFile instead of returning it.
+func (f *NamedFileContent) Decode(ctx *kong.DecodeContext) error { // nolint: golint
+	var filename string
+	err := ctx.Scan.PopValueInto("filename", &filename)
+	if err != nil {
+		return errors.Errorf("Failed to retrieve filename flag: %v", err)
+	}
+	// This allows unsetting of file content flags.
+	if filename == "" {
+		*f = NamedFileContent{}
+		return nil
+	}
+	filename = kong.ExpandPath(filename)
+	data, err := ioutil.ReadFile(filename) // nolint: gosec
+	if err != nil {
+		// Instead of returning an error, set contents to nil and store the error.
+		// The error will be checked during validation when building the configuration.
+		f.Contents = nil
+		f.Error = err
+	} else {
+		f.Contents = data
+	}
+
+	f.Filename = filename
+	return nil
+}
+
+// validateTLSFiles returns the first error it finds from the NamedFileContent fields. The errors are
+// populated by kong during the decode step, and are evaluated during validation.
+func (t *TLSConfig) validateTLSFiles() error {
+	if t.CACert.Error != nil {
+		return errors.Errorf("failed to open %q: %s", t.CACert.Filename, t.CACert.Error)
+	}
+
+	if t.Key.Error != nil {
+		return errors.Errorf("failed to open %q: %s", t.Key.Filename, t.Key.Error)
+	}
+
+	if t.Cert.Error != nil {
+		return errors.Errorf("failed to open %q: %s", t.Cert.Filename, t.Cert.Error)
+	}
+	return nil
+}
+
+// BuildTLSConfig loads CA certs and public/private key pair given options.
+//
+// Will return (nil, nil) if both cert and keyare not specified.
+func BuildTLSConfig(config TLSConfig) (*tls.Config, error) {
```

Comment:
I don't think this belongs here - it seems to be converting the TLS config struct into some other specific tls config struct required by some other library.

Reply:
This converts the TLS config in the PranaDB configuration (cert files) to [Golang's crypto/tls Config](https://pkg.go.dev/crypto/tls#Config). This is then passed to the grpc TLS credentials configuration:

```
tlsConfig, err := tls.BuildTLSConfig(s.tlsConfig)
	if err != nil {
		return nil, errors.WithStack(err)
	}
	opts := []grpc.ServerOption{
		grpc.Creds(credentials.NewTLS(tlsConfig)),
	}
```

Reply:
We should move this into where it used as it's specific to the user, not the config itself.